src/browser/BLECentralPlugin.js

```diff
@@ -1,63 +1,194 @@
+   
 function notSupported() {
     console.log('BLE is not supported on the browser');
 }
 
+
+function getDeviceId(device_id) {
+    return device_id.startsWith('0x') ? parseInt(device_id) : device_id;
+}
+
+
 module.exports = {
+
+    devices: new Map(),
+    scanning: null,
+
     scan: function(services, seconds, success, failure) {
-        notSupported();
-        if (failure) failure();
-    },
-    startScan: function(services, success, failure) {
-        notSupported();
-        if (failure) failure();
+
+        if (!navigator.bluetooth) {
+            failure('Bluetooth is not supported on this browser.');
+            return;
+        }
+
+        this.scanning = true;
+
+        let options = {};
+
+        if (services && services.length) {
+            options.filters = [{
+                services: services.map(getDeviceId)
```

Comment:
It's a little confusing to have `getDeviceId` being used on a service UUID like this. Perhaps `parseUUID` would be a better function name?